lib/cadence/activity/task_processor.rb

```diff
@@ -79,6 +79,24 @@ def respond_failed(reason, details)
         Cadence.logger.error("Unable to fail Activity #{activity_name}: #{error.inspect}")
         Cadence::ErrorHandler.handle(error, metadata: metadata)
       end
+
+      def deserialize(input)
+        JSON.deserialize(task.input)
+      rescue Oj::ParseError
+        # cadence official go-client serializes / deserializes input in a different format than this ruby client
+        # adding additional deserialization logic here to help read input that is passed from go-client
+        #
+        # this ruby client serializes / deserializes everything as one big string like blow:
+        # [1012474654, "second input"]
+        #
+        # while go client serializes input as separate input followed by line break
```

Comment:
Can we link the go client serialization link here for reference?

Reply:
done